safenode/src/node/api.rs

```diff
@@ -116,10 +123,23 @@ impl Node {
                             None => error!("The `TransferAction` channel is closed")
                         }
                     }
+                    _ = tokio::time::sleep(inactivity_timeout) => {
+                        let random_target = NetworkAddress::from_peer(PeerId::random());
+
+                        debug!("No network activity in the past {inactivity_timeout:?}, performing a random get_closest query to target: {random_target:?}");
+                        if let Ok(closest) = network_clone.node_get_closest_peers(&random_target).await {
+                            debug!("Network inactivity: get_closest returned {closest:?}");
+                        }
+                    }
                 }
             }
         });
 
+        // perform a get_closest query to self on node join. This should help populate the node's RT
+        let _ = network
```

Comment:
At this point, `network` instance may not discover any peer yet?
I think this query shall only got fired on the first time of received `PeerAdded`/`ConnectedToNetwork` ?

Reply:
Sorry, you're right. Got `Not enough peers in the k-bucket to satisfy the request`. Updating it.